diff --git a/embodied/jax/agent.py b/embodied/jax/agent.py
index ba90423..74b7893 100644
--- a/embodied/jax/agent.py
+++ b/embodied/jax/agent.py
@@ -172,13 +172,13 @@ class Agent(embodied.Agent):
 
     self._split = jax.jit(
         lambda xs: jax.tree.map(lambda x: list(x), xs),
-        internal.local_sharding(self.policy_sharded),
-        internal.local_sharding(self.policy_mirrored))
+        in_shardings=internal.local_sharding(self.policy_sharded),
+        out_shardings=internal.local_sharding(self.policy_mirrored))
     self._stack = jax.jit(
         lambda xs: jax.tree.map(
             jnp.stack, xs, is_leaf=lambda x: isinstance(x, list)),
-        internal.local_sharding(self.policy_mirrored),
-        internal.local_sharding(self.policy_sharded))
+        in_shardings=internal.local_sharding(self.policy_mirrored),
+        out_shardings=internal.local_sharding(self.policy_sharded))
 
     self._ckpt_groups = internal.grouped_ckpt_fns(
         self.params, self.jaxcfg.ckpt_chunksize)
diff --git a/embodied/jax/internal.py b/embodied/jax/internal.py
index ae9047b..cfc5339 100644
--- a/embodied/jax/internal.py
+++ b/embodied/jax/internal.py
@@ -277,9 +277,11 @@ def ckpt_fn(params, compile=True):
   keys = params.keys()
   original = {k: params[k].sharding for k in keys}
   inspec = {k: struct(params[k], original[k]) for k in keys}
-  gather_fn = jax.jit(lambda x: x, (original,), mirrored).lower(inspec)
+  gather_fn = jax.jit(
+      lambda x: x, in_shardings=(original,), out_shardings=mirrored).lower(inspec)
   inspec = {k: struct(params[k], mirrored) for k in keys}
-  shard_fn = jax.jit(lambda x: x, (mirrored,), original).lower(inspec)
+  shard_fn = jax.jit(
+      lambda x: x, in_shardings=(mirrored,), out_shardings=original).lower(inspec)
   if compile:
     gather_fn = gather_fn.compile()
     shard_fn = shard_fn.compile()
diff --git a/embodied/jax/transform.py b/embodied/jax/transform.py
index bc1b27e..40bd140 100644
--- a/embodied/jax/transform.py
+++ b/embodied/jax/transform.py
@@ -53,7 +53,9 @@ def init(
   if print_partition:
     print_grouping(grouping)
 
-  fn = jax.jit(fn, arg_shardings, params_sharding, static_argnums, None)
+  fn = jax.jit(
+      fn, in_shardings=arg_shardings, out_shardings=params_sharding,
+      static_argnums=static_argnums, static_argnames=None)
   params = fn(*dummy_inputs)
 
   return params, params_sharding
@@ -129,7 +131,9 @@ def apply(
         nn.LAYER_CALLBACK = old
       return outs
 
-  fn = jax.jit(fn, in_shardings, out_shardings, static_argnums, None, donate)
+  fn = jax.jit(
+      fn, in_shardings=in_shardings, out_shardings=out_shardings,
+      static_argnums=static_argnums, static_argnames=None, donate_argnums=donate)
 
   return fn
 
